library/core/src/tuple.rs

```diff
@@ -5,66 +5,71 @@ use crate::cmp::*;
 
 // macro for implementing n-ary tuple functions and operations
 macro_rules! tuple_impls {
-    ( $( ( $( $T:ident )+ ) )+ ) => {
-        $(
-            #[stable(feature = "rust1", since = "1.0.0")]
-            impl<$($T:PartialEq),+> PartialEq for ($($T,)+) where last_type!($($T,)+): ?Sized {
-                #[inline]
-                fn eq(&self, other: &($($T,)+)) -> bool {
-                    $( ${ignore(T)} self.${index()} == other.${index()} )&&+
-                }
-                #[inline]
-                fn ne(&self, other: &($($T,)+)) -> bool {
-                    $( ${ignore(T)} self.${index()} != other.${index()} )||+
-                }
+    ($T:ident) => {
+        tuple_impls!(@impl $T);
+    };
+    ($T:ident $( $U:ident )*) => {
+        tuple_impls!(@impl $T $( $U )*);
+        tuple_impls!($( $U )*);
```

Comment:
Could you explicitly say in a comment that recursively calling this macro on the tuple tail provides impls for tuples with lesser arity? This kind of recursive code is almost always harder to read and understand (but not by much in this specific case).

Reply:
Done. Let me know if there's anything missing